src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -27,29 +28,55 @@ public static bool HasErrors(this IOperation operation, Compilation compilation,
             return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.DefaultSeverity == DiagnosticSeverity.Error);
         }
 
+        private static readonly ObjectPool<Stack<IEnumerator<IOperation>>> s_childEnumeratorStackPool =
+            new ObjectPool<Stack<IEnumerator<IOperation>>>(() => new Stack<IEnumerator<IOperation>>(), 10);
```

Comment:
Can use `ArrayBuilder` rather than `Stack`. `ArrayBuilder` has an existing pool and has `Push` and `Pop` methods.